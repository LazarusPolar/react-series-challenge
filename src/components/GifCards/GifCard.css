.GifCards {
    z-index: 10px;
    display: inline-block;
}

.GifCards img {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}