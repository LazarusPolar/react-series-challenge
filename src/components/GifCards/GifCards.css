.GifCards {
    z-index: 10px;
    display: inline-block;
}

.GifCards img {
    width: 250px;
    height: 150px;
    z-index: 10px;
    display: inline-block;
}