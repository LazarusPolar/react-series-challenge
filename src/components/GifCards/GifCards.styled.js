import styled from 'styled-components';

// Super awesome box shadow effect provided from W3Schools
// Awesome job link: https://www.w3schools.com/cssref/tryit.asp?filename=trycss3_box-shadow4
export const GifCardsStyled = styled.div`
    display: inline-block;
    background-color: ${props => props.theme.favoritePageBackground};
    padding: 10px;
    margin: 10px;
    border: 1px solid;
    padding: 10px;
    box-shadow: 5px 5px 8px blue, 10px 10px 8px red, 15px 15px 8px green;
`

export const StyledH1 = styled.h1`
    color: #ffffff;
    font-family: 'Quicksand', sans-serif;  
`