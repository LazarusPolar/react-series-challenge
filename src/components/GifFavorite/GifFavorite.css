.GifFavorite {
    z-index: 10px;
    display: inline-block;
}

.GifFavorite img {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}