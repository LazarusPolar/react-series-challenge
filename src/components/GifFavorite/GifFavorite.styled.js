import styled from 'styled-components';

export const StyledGifFavorite = styled.div`
    text-align: center;
    padding-left: 0px;
    padding-right: 2px;
    width: 100%;
    background-color: ${props => props.theme.cardBackground};
`

