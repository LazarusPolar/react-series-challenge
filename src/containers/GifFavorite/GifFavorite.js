import React, { Component } from 'react';

class GifFavorite extends Component {

}

export default GifFavorite;