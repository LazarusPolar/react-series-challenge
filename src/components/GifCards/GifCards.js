import React from 'react';
import GifCard from '../GifCards/GifCard/GifCard'
import GifCardControls from '../GifCards/GifCardControls/GifCardControls'
import './GifCards.css';

const GifCards = (props) => {

        let arrays = null;
        const { gifData, searchedValue, searchedGifs } = props; 
        //console.log("props", props); 

        if(gifData !== null && gifData !== undefined){
            arrays = Object.keys(gifData).map((key, index) => {
                let { id, url } = gifData[index];

                console.log("url: " + url);
                //return (<img key = {id} src = {url} alt = "Cargando..." />);
                return (
                <div key = {id} className = "GifCards">
                <GifCard 
                    //loadError = {props.loadError}
                    //loadSuccess = {props.loadSuccess}
                    imageUrl = {url}
                     />
                    <GifCardControls key = {index} searchedGifs = {() => searchedGifs({ id: id, searchedValue: searchedValue, url: url})} />
                </div>);
            });

            //console.log("Arreglo" , arrays[0]);
            if(arrays.length === 0){
                arrays = <h1> No results :( </h1>;
            }
        }

        /*
        if(this.state.isLoading) {
            image = <p> Cargando imagen... </p>
        } else {
            //TODO: Spinner
            loading = <p> Cargando </p>
            image = <img src = {this.props.imageUrl} onLoad = {this.handleOnLoad} onError = {this.handleError} />
        }*/

        return(
            <div className = "GifCards">
                { arrays }
            </div>
        )
    
}

export default GifCards;