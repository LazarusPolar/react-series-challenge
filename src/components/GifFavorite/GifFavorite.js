import React, { Component } from 'react';
import SearchBar from '../../UI/SearchBar';
import GifCards from '../GifCards/GifCards';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchFunctions from '../../actions/searchValues';

class GifFavorite extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchedValue: null, 
            searchResults: props.favoritedImages, 
        }
    }

    updateState = (event) => {
        console.log(event);
        this.setState({ searchedValue: event })
    }

    handleSearch = () => {
        console.log(this.props.favoritedImages);
        //const searchParam = event.target.value;
        let newArray = Object.keys(this.props.favoritedImages).map((key, index) => {
            return this.props.favoritedImages[index]
        }).filter(word => {
            console.log("WORD", word);
            console.log("SEARCHEDVALUE", this.state.searchedValue)
            return word.searchedValue.includes(this.state.searchedValue)});
        
        console.log("NUEVO ARRAY", newArray);

        if(newArray !== undefined && newArray !== null && newArray.length >= 0){
            console.log("Entro a actualizar");
            this.setState({ searchResults: newArray });
        } 
        
    }

    render() {
        return (
            <div className = "GifFavorite">
                <SearchBar 
                updateState = { this.updateState } 
                handleSearch = { this.handleSearch } />
    
                <GifCards 
                gifData = { this.state.searchResults } 
                searchedGifs = { this.props.searchedFavoriteGifs } 
                searchedValue = { this.props.searchedValue }/>
            </div>
        );
    }

}

const mapStateToProps = (state) => {
    console.log(state);
    const { greeting, favoritedImages } = state;
    return {
        greeting, favoritedImages
    }
}

const mapDispatchToProps = (dispatch) => {
    const { searchedFavoriteGifs, searchedTrendingGifs, searchedSpecifiedGifs, updateSearchValue } = SearchFunctions.creators;
    return bindActionCreators( { searchedFavoriteGifs, searchedTrendingGifs, searchedSpecifiedGifs, updateSearchValue }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(GifFavorite);