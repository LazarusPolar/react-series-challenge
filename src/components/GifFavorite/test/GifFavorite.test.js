import GifCards from '../../GifCards/GifCards';
import ErrorBoundary from '../../../ErrorBoundary/ErrorBoundary'
import GifFavorite from '../GifFavorite'
import { StyledGifFavorite } from '../GifFavorite.styled'
import SearchBar from '../../../UI/SearchBar'
import React from 'react';
import { shallow } from 'enzyme';

describe('GifFavorite', () => {
    const func = jest.fn();
    const searchResult = [{
        id: 1,
        searchedValue: "trendy",
        url: "https://image", 
        isFavorite: true,
    }]
    const searchResultEmpty = [];
    const themeSunny = {
        gifPageBackground: '#969ec3',
        favoritePageBackground: '#c8bed7',
        headerBackground: '#fde1cc',
        cardBackground: '#fec492',
    }
    let wrapper;
/*
    beforeEach(() => {
        wrapper = shallow(
            <GifFavorite />
        );
    });

    test('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    
    test('should set GifCards ', () => {
        expect(wrapper.contains(
            <StyledGifFavorite {...themeSunny}>
                    <SearchBar 
                    updateState = { func } 
                    handleSearch = { func } />
                <ErrorBoundary>
                    <GifCards 
                    gifData = { searchResult } 
                    addRemoveGif = { func } 
                    searchedValue = { func }/>
                </ErrorBoundary>
            </StyledGifFavorite>
        )).toBe(true);
    });*/

})
