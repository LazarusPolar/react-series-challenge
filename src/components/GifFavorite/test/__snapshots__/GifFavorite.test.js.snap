// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifFavorite should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GifFavorite />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <SearchBar
          handleSearch={[Function]}
          updateState={[Function]}
/>,
        <ErrorBoundary>
          <GifCards
                    addRemoveGif={undefined}
                    gifData={undefined}
                    searchedValue={undefined}
          />
</ErrorBoundary>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSearch": [Function],
          "updateState": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <GifCards
            addRemoveGif={undefined}
            gifData={undefined}
            searchedValue={undefined}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addRemoveGif": undefined,
            "gifData": undefined,
            "searchedValue": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <SearchBar
            handleSearch={[Function]}
            updateState={[Function]}
/>,
          <ErrorBoundary>
            <GifCards
                        addRemoveGif={undefined}
                        gifData={undefined}
                        searchedValue={undefined}
            />
</ErrorBoundary>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSearch": [Function],
            "updateState": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <GifCards
              addRemoveGif={undefined}
              gifData={undefined}
              searchedValue={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "addRemoveGif": undefined,
              "gifData": undefined,
              "searchedValue": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
